java/vector/src/test/java/org/apache/arrow/vector/file/BaseFileTest.java

```diff
@@ -408,28 +410,66 @@ protected void validateNestedDictionary(VectorSchemaRoot root, DictionaryProvide
     Assert.assertEquals(new Text("bar"), dictionaryAccessor.getObject(1));
   }
 
-  protected void validateNestedDictionary(ListVector vector, DictionaryProvider provider) {
-    Assert.assertNotNull(vector);
-    Assert.assertNull(vector.getField().getDictionary());
-    Field nestedField = vector.getField().getChildren().get(0);
+  private void writeIntAsDecimal(NullableDecimalVector vector, long value, int index) {
+    ArrowType.Decimal type = (ArrowType.Decimal) vector.getField().getType();
+    BigDecimal decimalValue = new BigDecimal(BigInteger.valueOf(value), type.getScale());
+    DecimalUtility.writeBigDecimalToArrowBuf(decimalValue, vector.getBuffer(), index);
+    vector.getValidityVector().getMutator().setToOne(index);
```

Comment:
This procedure to write BigDecimal values to a vector is pretty awkward

Reply:
I agree. I prefer that we add `setSafe(index, decimalValue)`